assets/js/base/hooks/checkout/use-checkout-submit.js

```diff
@@ -1,12 +1,39 @@
 /**
  * External dependencies
  */
-import { useCheckoutContext } from '@woocommerce/base-context';
+import { __ } from '@wordpress/i18n';
+import {
+	useCheckoutContext,
+	usePaymentMethodDataContext,
+} from '@woocommerce/base-context';
+import { usePaymentMethods } from '@woocommerce/base-hooks';
 
 /**
- * Returns the submitLabel and onSubmit interface from the checkout context
+ * Returns the submitButtonText, onSubmit interface from the checkout context,
+ * and an indication of submission status.
  */
 export const useCheckoutSubmit = () => {
-	const { submitLabel, onSubmit } = useCheckoutContext();
-	return { submitLabel, onSubmit };
+	const {
+		onSubmit,
+		isCalculating,
+		isBeforeProcessing,
+		isProcessing,
+		isAfterProcessing,
+		isComplete,
+		hasError,
+	} = useCheckoutContext();
+	const { paymentMethods } = usePaymentMethods();
+	const { activePaymentMethod } = usePaymentMethodDataContext();
+	const paymentMethod = paymentMethods[ activePaymentMethod ] || {};
```

Comment:
No need to change in this pull, but just got thinking I wonder if it'd be useful to expose the `activePaymentMethodObject` via `usePaymentMethods` (which would internally consume `usePaymentMethodDataContext` to expose it)? It could reduce some code in place likely.

Reply:
yes! I was looking for this but it didn't exist.